spec/javascripts/Dashboard/components/PotentialUpgradesWidget.spec.jsx

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import React from 'react'
+
+import { PotentialUpgradesWidget } from 'Dashboard/components/PotentialUpgradesWidget'
+import { mount } from 'enzyme'
+
+let props
+const defaultProps = {
+  violations: [
+    {
+      id: 1,
+      account_id: 1,
+      account_name: 'Account Name',
+      alerts_count: 10,
+      url: '/apiconfig/alerts?search%5Baccount_id%5D=1&search%5Blevel%5D=100&search%5Btimestamp%5D%5B%5D=2022-09-24&search%5Btimestamp%5D%5B%5D=2022-10-24'
+    }
+  ],
+  incorrectSetUp: true,
+  links: {
+    adminServiceApplicationPlans: '/apiconfig/services/1/application_plans',
+    settingsAdminService: '/apiconfig/services/1/settings#web_provider'
+  }
+}
+
+const mountWrapper = (props) => mount(<PotentialUpgradesWidget {...{ ...defaultProps, ...props }}/>)
+
+it('should render', () => {
+  const wrapper = mountWrapper(defaultProps)
+  expect(wrapper.exists()).toBe(true)
+})
+
+describe('when the setup is incorrect', () => {
+  const wrapper = mountWrapper(defaultProps)
+
+  it('should render incorrect setup messages', () => {
+    expect(wrapper.text().includes('In order to show Potential Upgrades, add 1 or more usage limits to')).toBe(true)
+    expect(wrapper.text().includes('Web Alerts for Admins of this Account of 100%')).toBe(true)
+  })
+})
+
+describe('when the setup is correct', () => {
+  const correctSetUpProps = { ...defaultProps, incorrectSetUp: false }
+
+  describe('and there are violations', () => {
+    const wrapper = mountWrapper(correctSetUpProps)
```

Comment:
Same as with the other `spec`: I'd include this line (and applies to every case where the `wrapper` is built this way) inside every `test` rather than inside the `describe` or I'd use a `beforeEach` to create the `wrapper` and an `afterEach` to clean it.

Reply:
Good! Added in 4b1b900.